### PR TITLE
[minizinc] Initial Fuzzer

### PR DESCRIPTION
**Pull request by Dekker1**
This PR adds the initial fuzzer and seed corpus for the MiniZinc compiler

### CODE REVIEW

**Comment by harsilspatel** [comment]
@googlebot I fixed it.

**Comment by harsilspatel** [comment]
@googlebot I consent.

**Comment by inferno-chromium** [comment]
@jonathanmetzman - can you help to review.

File: projects/minizinc/minizinc_fuzzer.cpp [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    Timer starttime;
+    bool fSuccess = false;
+    try {
+      MznSolver slv(std::cout,std::cerr);
+      try {
+        std::string model = std::string(reinterpret_cast<const char*>(Data), Size);
+        std::vector<std::string> args({"-c","--solver","org.minizinc.mzn-fzn"});
+        fSuccess = (slv.run(args, model, "minizinc", "model.mzn") != SolverInstance::ERROR);
+      } catch (const LocationException& e) {
+        if (slv.get_flag_verbose())
```
Comment (jonathanmetzman) [inline projects/minizinc/minizinc_fuzzer.cpp]: This wont log heavily when run on OSS-Fuzz right?

File: projects/minizinc/minizinc_fuzzer.dict [hunk 2]
```diff
@@ -0,0 +1,50 @@
+reserved_keyword_0="ann"
```
Comment (jonathanmetzman) [inline projects/minizinc/minizinc_fuzzer.dict]: tip: you don't actually need to use the X="Y" syntax with dictionaries.
"Y" is sufficient.

File: projects/minizinc/project.yaml [hunk 3]
```diff
@@ -7,4 +7,3 @@ sanitizers:
   - address
   - memory:
      experimental: True
-  - undefined
```
Comment (jonathanmetzman) [inline projects/minizinc/project.yaml]: Why doesn't UBSAN work?

File: projects/minizinc/minizinc_fuzzer.cpp [hunk 4]
```diff
@@ -0,0 +1,48 @@
+#include <minizinc/solver.hh>
```
Comment (jonathanmetzman) [inline projects/minizinc/minizinc_fuzzer.cpp]: This file needs a license header as well.

**Comment by inferno-chromium** [comment]
@Dekker1 - it is preferred that you enable MSan and UBSan in later PRs. MSan should just work if you don't have system lib dependencies (note that experimental:true would mean no bug filing, so discouraged).
Also, please check https://github.com/google/oss-fuzz/blob/master/docs/ideal_integration.md, we prefer than fuzz target file and build.sh should ideally go into your repo.

File: projects/minizinc/minizinc_fuzzer.cpp (hunk 1 above)
Comment (Dekker1) [reply projects/minizinc/minizinc_fuzzer.cpp]: This is code copied form the main MiniZinc executable. When it is executed in verbose mode extra line breaks are added. The fuzzer, however, is never executed in verbose mode.
The only output that the fuzzer should give are the errors produced by the compiler whenever an error is found.

File: projects/minizinc/project.yaml (hunk 3 above)
Comment (Dekker1) [reply projects/minizinc/project.yaml]: We're currently unsure why it is not working, but it is failing the CI tests. We hope to add it back soon.

**Comment by Dekker1** [comment]
Thank you for the detailed feedback! We'll have a look and hope to make some changes to improve the fuzzer soon.

**Pull request merged**